From 3981805ec78adf4479f0988cfc2174a02d58a738 Mon Sep 17 00:00:00 2001
From: "Carlos R. Mafra" <crmafra@gmail.com>
Date: Thu, 10 Apr 2014 19:34:48 +0100
Subject: [PATCH] wmgenmenu: Write paths according to options set at configure
 time

As reported by Shawn W Dunn, the configuration strings written by wmgenmenu
in $HOME/GNUstep/Defaults/WMRootMenu were not reflecting his installation
directories choices.

Fix this by writing strings composed with PKGDATADIR.

Signed-off-by: Carlos R. Mafra <crmafra@gmail.com>
---
Index: util/wmgenmenu.c
===================================================================
--- util/wmgenmenu.c.orig
+++ util/wmgenmenu.c
@@ -41,12 +41,13 @@ int main(int argc, char *argv[])
 {
 	char *t;
 	int ch;
-	char *tmp, *theme_paths, *style_paths, *icon_paths;
+	char *tmp, *theme_paths, *style_paths, *icon_paths, *bg_paths;
 
 	tmp = wstrconcat("-noext ", PKGDATADIR);
 	theme_paths = wstrconcat(tmp, "/Themes $HOME/" GSUSER_SUBDIR "/" USERDATA_SUBDIR "/" PACKAGE_TARNAME "/Themes WITH setstyle");
 	style_paths = wstrconcat(tmp, "/Styles $HOME/" GSUSER_SUBDIR "/" USERDATA_SUBDIR "/" PACKAGE_TARNAME "/Styles WITH setstyle");
 	icon_paths = wstrconcat(tmp, "/IconSets $HOME/" GSUSER_SUBDIR "/" USERDATA_SUBDIR "/" PACKAGE_TARNAME "/IconSets WITH seticons");
+  bg_paths = wstrconcat(tmp, "/Backgrounds $HOME/" GSUSER_SUBDIR "/" USERDATA_SUBDIR "/" PACKAGE_TARNAME "/Backgrounds WITH wmsetbg -u -t");
 
 	struct option longopts[] = {
 		{ "version",		no_argument,	NULL,	'v' },
@@ -245,7 +246,7 @@ int main(int argc, char *argv[])
 	L3Menu = WMCreatePLArray(
 		WMCreatePLString(_("Images")),
 		WMCreatePLString("OPEN_MENU"),
-		WMCreatePLString("-noext $HOME/" GSUSER_SUBDIR "/" USERDATA_SUBDIR "/" PACKAGE_TARNAME "/Backgrounds WITH wmsetbg -u -t"),
+		WMCreatePLString(bg_paths),
 		NULL
 	);
 	WMAddToPLArray(L2Menu, L3Menu);
@@ -334,7 +335,7 @@ int main(int argc, char *argv[])
 	L1Menu = WMCreatePLArray(
 		WMCreatePLString(_("Configure Window Maker")),
 		WMCreatePLString("EXEC"),
-		WMCreatePLString("WPrefs"),
+		WMCreatePLString("/usr/lib/GNUstep/Applications/WPrefs.app/WPrefs"),
 		NULL
 	);
 	WMAddToPLArray(RMenu, L1Menu);
